custom_components/rct_power/lib/state_helpers.py

```diff
@@ -60,3 +61,48 @@ def sum_api_response_values_as_state(
         for value in values
         if isinstance(value, (int, float))
     )
+
+
+def get_first_api_response_value_as_binary_state(
+    entity: BinarySensorEntity,
+    values: list[Optional[ApiResponseValue]],
+) -> Union[None, bool]:
+    if len(values) <= 0:
+        return None
+
+    return get_api_response_value_as_binary_state(entity=entity, value=values[0])
+
+
+def get_api_response_value_as_binary_state(
+    entity: BinarySensorEntity,
+    value: Optional[ApiResponseValue],
+) -> Union[None, bool]:
+    if value is None:
+        return None
+    return bool(value)
+
+
+def get_battery_calibration_status(
+    entity: BinarySensorEntity,
+    values: list[Optional[ApiResponseValue]],
+) -> Union[None, bool]:
+    if len(values) <= 0:
+        return None
+    value = values[0]
+
+    if isinstance(value, int):
+        return value & 1032 != 0
```

Comment:
Did you mean `1024` or is this actually a combination of flags? And I'd be interested to learn how you found out which flag means what.

Reply:
To answer myself: using `1032` as a binary mask is what the official app does too :shrug: